site24x7/websitemonitor.go

```diff
@@ -207,199 +238,72 @@ func websiteMonitorCreateOrUpdate(method, url string, expectedResponseStatus int
 	}
 
 	i := 0
-	actionRefs := make([]ActionRef, len(d.Get("actions").(map[string]interface{})))
-	for k, v := range d.Get("actions").(map[string]interface{}) {
+	actionRefs := make([]api.ActionRef, len(d.Get("action_ids").(map[string]interface{})))
+	for k, v := range d.Get("action_ids").(map[string]interface{}) {
 		tmp, err := strconv.Atoi(k)
 		if err != nil {
 			panic(err)
```

Comment:
We should not panic here. This makes the code fragile and hard to test. Make the `resourceDataToWebsiteMonitor` func return an error as the 2nd return value instead and pass it through up the stack.

The `panic` is also the reason why your tests were crashing.